### PR TITLE
Add Treegrid Example

### PR DESCRIPTION
**Pull request by mcking65**
For issue #132, merging review-ready version of treegrid example developed by @aleventhal and edited to fit example template by @jongund.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/762.html" title="Last updated on Jul 12, 2018, 6:51 AM GMT (514f612)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/762/a49121c...514f612.html" title="Last updated on Jul 12, 2018, 6:51 AM GMT (514f612)">Diff</a>

### CODE REVIEW

File: examples/treegrid/treegrid-1.html [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+    <div id="ex1">
+      <table id="treegrid" role="treegrid" aria-label="ARIA treegrid e-mail inbox example">
+        <colgroup>
+          <col id="treegrid-col1">
+          <col id="treegrid-col2">
+          <col id="treegrid-col3">
+        </colgroup>
+        <thead>
+          <tr>
+            <th role="columnheader">Subject</th>
```
Comment (mcking65) [inline examples/treegrid/treegrid-1.html]: @aleventhal, @jongund, why is role columnheader added here. This is not necessary and not recommended to duplicate native semantics. Can we safely remove without breaking any selectors or js?

File: examples/treegrid/treegrid-1.html [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+            <th role="columnheader">Summary</th>
```
Comment (mcking65) [inline examples/treegrid/treegrid-1.html]: @aleventhal, @jongund, why is role row added here and on subsequent tr elements? This is not necessary and not recommended to duplicate native semantics. Can we safely remove without breaking any selectors or js?

File: examples/treegrid/treegrid-1.html (hunk 1 above)
Comment (aleventhal) [reply examples/treegrid/treegrid-1.html]: Actually, a th can be a columnheader or a rowheader. I don't think it's redundant to specify. This way the browser does not need to fall back on heuristics (which different browsers may disagree about).

Comment (aleventhal) [reply examples/treegrid/treegrid-1.html]: Actually we can probably use scope="col".

File: examples/treegrid/treegrid-1.html (hunk 2 above)
Comment (aleventhal) [reply examples/treegrid/treegrid-1.html]: Good question. I would guess that it makes no difference and can be removed, but I think it should be tested with at least one screen reader to be completely sure.

FWIW, I find it easier to understand the examples when the roles are explicitly stated.  When we remove "row" from the markup, should we also make sure it's clear in the docs that <tr> can be replaced with something like <div role="row"> ?

**Comment by mcking65** [comment]
OK ... Editorially in reasonable shape and the only errors left are the posinset and setsize errors, which are being addressed by w3c/aria#558. So, merging to master.

**Pull request merged**